src/DocumentFormat.OpenXml/Packaging/OpenXmlPackage.FlatOpc.cs

```diff
@@ -49,68 +54,85 @@ protected XDocument ToFlatOpcDocument(XProcessingInstruction instruction)
             // to call it here.
             Save();
 
+            // Identify all AlternativeFormatInputParts (AltChunk parts).
+            // This is necessary because AltChunk parts must be treated as binary
+            // parts regardless of the actual content type, which might even be
+            // XML-related such as application/xhtml+xml.
+            var altChunkPartUris = new HashSet<Uri>(
+                new InternalPackageRelationships(Package)
+                    .Where(pr => pr.RelationshipType == AltChunkRelationshipType)
+                    .Select(pr => PackUriHelper.ResolvePartUri(pr.SourceUri, pr.TargetUri)));
+
             // Create an XML document with a standalone declaration, processing
             // instruction (if not null), and a package root element with a
             // namespace declaration and one child element for each part.
             return new XDocument(
                 new XDeclaration("1.0", "UTF-8", "yes"),
                 instruction,
                 new XElement(
-                    pkg + "package",
-                    new XAttribute(XNamespace.Xmlns + "pkg", pkg.ToString()),
-                    Package.GetParts().Select(part => GetContentsAsXml(part))));
+                    Pkg + "package",
+                    new XAttribute(XNamespace.Xmlns + "pkg", Pkg.ToString()),
+                    Package.GetParts().Select(p => GetContentsAsXml(p, altChunkPartUris))));
```

Comment:
I personally do not like abbreviations in code but I don't know about the style guide of this project. So that's up to you...

Reply:
We might even be able to fit in `art` twice ;-)